include/tvm/topi/transform.h

```diff
@@ -1511,6 +1511,35 @@ inline Tensor sparse_to_dense(const Tensor& sparse_indices, const Array<Integer>
       name, tag);
 }
 
+/*!
+ * \brief Returns a tensor with the diagonal of input tensor replaced with the provided diagonal.
+ * \param input input tensor.
+ * \param diagonal values to be filled in the diagonal.
+ * \param name output tensor name.
+ * \param tag output tensor tag.
+ * \return new tensor with given diagonal values.
+ */
+inline Tensor matrix_set_diag(const Tensor& input, const Tensor& diagonal,
```

Comment:
A suggestion:- may be if we can support `alignment` and `k`(offset) similar to `MatrixSetDiagV3` in tf, it will be good. we can support directly for tensorflow ops as well.

Reply:
That might take some time.
Would it be fine to have that in a follow-up PR?